draft-wang-ppm-differential-privacy.md

```diff
@@ -255,12 +255,58 @@ with the correct noise type.
 > TODO: Chris P: we will mention Prio3SumVec because that's what we use to describe aggregator DP with amplification
 
 
-# Differential Privacy Mechanism {#dp mechanism}
+# Differential Privacy Mechanism {#dp-mechanism}
 
 This section should define a framework to apply different dp noise type to a
 client measurement and/or an aggregation.
 
-> TODO: Junye to spell out the interface of dp mechanisms, "Noise Mechanisms to Accomplish DP"
+Clients and Aggregators can implement various noise mechanisms to accomplish
+the DP guarantees described in {{budget}}. The parameters of the noise
+mechanisms should be computed based on the target DP guarantee, and can be
+passed to the initializers of the noise mechanisms.
+
+We define three methods for the noise mechanism interface `DpMechanism`:
+
+* `add_noise` that adds noise to a piece of data. This can be used by a Client
+  to add noise to its measurement.
+
+* `sample_noise` that allows `DpMechanism` to sample noise directly. This can
+  be used by an Aggregator to sample Aggregator DP noise, which then gets merged
+  with the Aggregator's aggregate share.
+
+* `debias` that allows `DpMechanism` to debias the data due to the bias
+  generated by the noise. Note not all noise will need this functionality.
+  We currently think some local randomizers will need this functionality.
+  For example, Appendix A and Appendix C of {{MJTBCCDFGHJKLLMPPPPPRSWZ22}}
+  describe two types of randomized response local randomizers that will
+  need the debiasing functionality.
+
+~~~
+class DpMechanism:
+    DataType
+    DebiasedDataType
+
+    def add_noise(self, data: DataType) -> DataType:
+        """Add noise to a piece of data. """
+        pass
+
+    def sample_noise(self, num_reps: int, dimension: int) -> DataType:
+        """
+        Sample noise for `num_reps` number of times,
+        with `dimension`.
+        """
+        pass
+
+    def debias(self,
```

Comment:
The set of local DP noises we have in mind don't remove noise completely, e.g. see Algorithm 2 and Algorithm 4 in [{{MJTBp22}}](https://arxiv.org/abs/2004.00010), the aggregate result will still have noise, with some standard deviation. I think `debias` is a better term because it is "removing" the positive bias added in the local DP.

Reply:
Ahh right, "remove noise" is not a good term. How about "remove bias"? The problem I'm trying to solve is that "debias" doesn't look like "de-bias" to me, it looks like "deebee-ass" :D

Reply:
Debias has been used in referenced literatures so it should be clear to readers, remove_bias would do too but I don’t know if we can avoid mentioning debias

Reply:
Sticking with `debias` for now. We can always change it to `remove_bias` if others feel it is better as well.